packages/backend-openapi-utils/src/middlewares/auditor.ts

```diff
@@ -0,0 +1,220 @@
+        regex: convertPathToRegex(path),
+        path,
+        pathItem,
+      });
+    }
+  }
+
+  function baseHandler(req: Request, res: Response): {
+    captureMetadata: () => JsonObject;
+    auditorConfig: AuditorExtension;
+  } | undefined {
+    // Find matching path
+    const requestPath = req.path;
+    const matcher = pathMatchers.find(m => m.regex.test(requestPath));
+
+    if (!matcher) {
+      return undefined;
+    }
+
+    // Get operation for the HTTP method
+    const method = req.method.toLowerCase();
+    const operation = matcher.pathItem[
+      method as keyof PathItemObject
+    ] as OperationObject;
+
+    if (!operation) {
+      return undefined;
+    }
+
+    // Check for x-backstage-auditor extension
+    const auditorConfig = operation['x-backstage-auditor'] as
+      | AuditorExtension
+      | undefined;
+
+    if (!auditorConfig) {
+      return undefined;
+    }
+
+    // Store the capture config for use in finish event
+    const { captureFromRequest, captureFromResponse, ...staticMeta } =
+      auditorConfig.meta || {};
+
+    // Capture metadata from request - some may not be available yet (params)
+    const captureMetadata = (): JsonObject => {
+      const meta: JsonObject = { ...staticMeta };
+
+      if (captureFromRequest) {
+        const { body, params, query } = captureFromRequest;
+
+        if (body) {
+          for (const field of body) {
+            const value = extractValueFromObject(req.body, field);
+            if (value !== undefined) {
+              meta[field] = value;
+            }
+          }
+        }
+
+        if (params) {
+          for (const field of params) {
+            const value = req.params[field];
+            if (value !== undefined) {
+              meta[field] = value;
+            }
+          }
+        }
+
+        if (query) {
+          for (const field of query) {
+            const value = req.query[field];
+            if (value !== undefined) {
+              meta[field] = value;
+            }
+          }
+        }
+      }
+
+      return meta;
+    };
+
+    return {captureMetadata, auditorConfig};
+  }
+
+  const success = (req: Request, res: Response, next: NextFunction) => {
+    const result = baseHandler(req, res);
+    if (!result) {
+      next();
+      return;
+    }
+    const { captureMetadata, auditorConfig } = result;
+
+    res.on('finish', () => {
+      if(res.statusCode >= 200 && res.statusCode < 300) {
+        // Create audit event - capture metadata after route matching
+        auditor
+          .createEvent({
+            eventId: auditorConfig.eventId,
+            severityLevel: auditorConfig.severityLevel,
+            meta: captureMetadata(),
+            request: req,
+          })
+          .then(auditorEvent => {
+            // Only mark as success for 2xx responses
+            // For errors, the handler should catch and call auditorEvent.fail()
+            if (res.statusCode >= 200 && res.statusCode < 300) {
```

Comment:
I believe this if statement is redundant because you check the status code on line 169.